### PR TITLE
TekScope2K - turn on HEADER and VERBOSE options to query available data sources correctly

### PR DESCRIPTION
**Pull request by RoaCode**
## Proposed changes

Please include a summary of the changes and any links to related issues. Please also include relevant motivation and context.

Addresses #326 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Functionality update (non-breaking change which updates or changes existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] CI/CD update (an update to the CI/CD workflows, scripts, and/or configurations)
- [ ] Documentation update (an update to enhance the user experience when reading through the docs)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] I have followed the guidelines in the [CONTRIBUTING](https://github.com/tektronix/tm_devices/blob/main/CONTRIBUTING.md) document
- [X] I have signed the CLA
- [X] I have checked to ensure there aren't other open [Pull Requests](https://github.com/tektronix/tm_devices/pulls) for the same update/change
- [X] I have created (or updated) an [Issue](https://github.com/tektronix/tm_devices/issues) to track the status of this update/change and updated the link in this PR description (see above in the **Proposed changes** section) using the wording `Addresses #<issue_number>`
- [X] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] Basic linting passes locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] I have added necessary documentation (if appropriate)
- [X] I have updated the Changelog with a brief description of my changes

### CODE REVIEW

**Review by nfelt14 — CHANGES_REQUESTED** [review]

File: src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py [hunk 1]
```diff
@@ -39,7 +39,14 @@ def get_available_data_sources(self) -> List[str]:
         Returns:
             The list of available data sources as strings.
         """
-        source_string = self.query("SELECT?")
+        self.write("HEADER 1")  # Setting HEADER = 1 breaks 'set_and_check'
```
[the remark this answers is not in the document]
Comment (RoaCode) [reply src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py]: Header setting is now queried and restored after 'SELECT?'. Verbose setting was not necessary, so removed.

File: src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py [hunk 2]
```diff
@@ -39,7 +39,11 @@ def get_available_data_sources(self) -> List[str]:
         Returns:
             The list of available data sources as strings.
         """
+        previous_header_state = self.query("HEADER?").split(" ")[-1]  # Read previous header state
+        self.write("HEADER 1")  # Turn on header state so SELECT query works correctly
         source_string = self.query("SELECT?")
+        self.write("HEADER" + previous_header_state)  # Return header state back to original
```
Comment (nfelt14) [inline src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py:45]:
```suggestion
        self.write(f"HEADER {previous_header_state}")  # Return header state back to original
```

**Review by nfelt14 — CHANGES_REQUESTED** [review]

**Comment by nfelt14** [comment]
@RoaCode you will need to rebase your branch to get builds to pass

**Review by nfelt14 — DISMISSED** [review]

**Comment by nfelt14** [comment]
@RoaCode, it looks like there are more conflicts with the changelog

**Review by nfelt14 — DISMISSED** [review]

**Comment by nfelt14** [comment]
@RoaCode your last few commits are unsigned

**Comment by nfelt14** [comment]
@RoaCode, it looks like this needs to be rebased again

File: CHANGELOG.md
Comment (nfelt14) [inline CHANGELOG.md:1]: This file is in a weird state. There are multiple sections duplicated, and the unreleased stuff is in multiple places.

**Review by nfelt14 — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -39,15 +39,23 @@ repos:
       - id: remove-tabs
       - id: forbid-tabs
   - repo: https://github.com/python-jsonschema/check-jsonschema
+<<<<<<< HEAD
```
Comment (nfelt14) [inline .pre-commit-config.yaml:42]: merge conflict

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,10 +18,9 @@ Valid subsections within a version are:
 
 Things to be included in the next release go here.
 
-### Added
+### Fixed
 
-- `collectgarbage()` is now called during cleanup of `TSPDevice` children.
```
Comment (nfelt14) [inline CHANGELOG.md:23]: Why did you remove these entries from the changelog?

**Review by nfelt14 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (RoaCode) [reply CHANGELOG.md:1]: Yes, I was struggling to get the unverified commits squashed and weird things have happened. I'm trying to get it sorted out.

File: CHANGELOG.md (hunk 4 above)
Comment (RoaCode) [reply CHANGELOG.md:23]: Not intentional. Hopefully those entries are back and everything is good.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (RoaCode) [reply .pre-commit-config.yaml]: I think I have this cleaned up

File: CHANGELOG.md
Comment (RoaCode) [reply CHANGELOG.md:1]: I think I got the CHANGELOG fixed now with all the entries.

**Review by nfelt14 — APPROVED** [review]
